### PR TITLE
Add ability to update status to incomplete

### PR DESCRIPTION
**Pull request by iain-bryson**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Looks ok to me; the real acid test will be running CI builds of Speckel against this mock.

File: lib/stripe_mock/request_handlers/subscriptions.rb [hunk 1]
```diff
@@ -219,12 +219,18 @@ def update_subscription(route, method_url, params, headers)
...
+        # https://github.com/stripe-ruby-mock/stripe-ruby-mock/issues/729
+        subscription = resolve_subscription_changes(subscription, subscription_plans, customer, params) unless subscription[:status] == 'incomplete'
 
         # delete the old subscription, replace with the new subscription
         customer[:subscriptions][:data].reject! { |sub| sub[:id] == subscription[:id] }
         customer[:subscriptions][:data] << subscription
 
+        if params[:status]
+          subscription[:status] = params[:status]
+        end
```
Comment (bobbybloor) [inline lib/stripe_mock/request_handlers/subscriptions.rb]:
```suggestion
        subscription[:status] = params[:status] if params[:status]
```

**Review by bobbybloor — APPROVED** [review]
LGTM, nonblocking comment

**Review by vishnun — APPROVED** [review]
Looks good. Can't tell if it'll have any side effects, let's give it a go.

**Pull request merged**